### DISCUSSION TITLE
Good practice about version control

### DISCUSSION THREAD

**Discussion by caenrique** [zk-org/zk#425] [General]
Hi, I started using zk and I'm loving it so far

I have a notes directory that I keep under version control with git, and I have created a zk notebook in it. Every time I modify something, the `.zk/notebook.db` is updated.  My question is:

Is it a good idea to commit the changes to this file with the notes? or there is a better way of doing things?

I'm guessing that if I don't keep that file, and checkout the repository in another machine, some of the links and stuff might break?

**Comment by tjex** [comment]
heyo, welcome!

You don't have to track changes in the notebook.db, but it won't create any 
issues if you do. 

But if you don't track it, and you want to revert a change or go back in 
history, you will at some point need to run `zk index`. Similarly, if you clone 
your Zettelkasten somewhere fresh, you won't clone down a notebook.db, and will 
also need to run `zk index` to get a fresh database.

**Reply by tjex** [reply]
To be clearer on your second question. Nothing will break, it's just that when you try to run any command for the first time in the cloned repo, like `zk list`, zk will run `zk index` to rebuild the notebook.db. Which will take some measure of time depending on how big your vault gets.

**Reply by tjex** [reply]
Also good to know as a newcomer, if you experience any weird behaviour when running queries against the notebook.db (dates are wrong, notes aren't being found, etc), you can first try to delete the notebook.db and run `zk index` again.

**Reply by caenrique** [reply]
Awesome thanks! 🙌
